python/pyspark/sql/connect/dataframe.py

```diff
@@ -115,6 +115,9 @@ def __init__(
         self._cache: Dict[str, Any] = {}
         self._session: "RemoteSparkSession" = session
 
+    def __repr__(self) -> str:
+        return "DataFrame[%s]" % (", ".join("%s: %s" % c for c in self.dtypes))
```

Comment:
this follows the default behavior of https://github.com/apache/spark/blob/40a9a6ef5b89f0c3d19db4a43b8a73decaa173c3/python/pyspark/sql/dataframe.py#L860-L869

Reply:
Question: is this public API?

Reply:
thanks for the example!